examples/zero-shot-object-detection/run_example.py

```diff
@@ -0,0 +1,117 @@
+    )
+    parser.add_argument(
+        "--use_hpu_graphs",
+        action="store_true",
+        help="Whether to use HPU graphs or not. Using HPU graphs should give better latencies.",
+    )
+    parser.add_argument(
+        "--bf16",
+        action="store_true",
+        help="Whether to use bf16 precision for classification.",
+    )
+    parser.add_argument(
+        "--print_result",
+        action="store_true",
+        help="Whether to print the classification results.",
+    )
+    parser.add_argument("--warmup", type=int, default=3, help="Number of warmup iterations for benchmarking.")
+    parser.add_argument("--n_iterations", type=int, default=5, help="Number of inference iterations for benchmarking.")
+
+    args = parser.parse_args()
+
+    adapt_transformers_to_gaudi()
+
+    processor = AutoProcessor.from_pretrained(args.model_name_or_path)
+    model = OwlViTForObjectDetection.from_pretrained(args.model_name_or_path)
+
+    image = Image.open(requests.get(args.image_path, stream=True).raw)
+    texts = []
+    for text in args.prompt.split(','):
+        texts.append(text)
+
+    if args.use_hpu_graphs:
+        model = ht.hpu.wrap_in_hpu_graph(model)
+
+    autocast = torch.autocast(device_type="hpu", dtype=torch.bfloat16, enabled=args.bf16)
+    model.to("hpu")
+
+    with torch.no_grad(), autocast:
+        for i in range(args.warmup):
+            inputs = processor(text=texts, images=image, return_tensors="pt").to("hpu")
+            outputs = model(**inputs)
+            torch.hpu.synchronize()
+
+        total_model_time = 0
+        for i in range(args.n_iterations):
+            inputs = processor(text=texts, images=image, return_tensors="pt").to("hpu")
+            model_start_time = time.time()
+            outputs = model(**inputs)
+            torch.hpu.synchronize()
+            model_end_time = time.time()
+            total_model_time = total_model_time + (model_end_time - model_start_time)
+
+            if args.print_result:
+                # Target image sizes (height, width) to rescale box predictions [batch_size, 2]
+                target_sizes = torch.Tensor([image.size[::-1]])
+
+                # Convert outputs (bounding boxes and class logits) to Pascal VOC format (xmin, ymin, xmax, ymax)
+                results = processor.post_process_object_detection(outputs=outputs, target_sizes=target_sizes, threshold=0.1)
+                if i == 0:
+                    text = texts[i] # Retrieve predictions for the first image for the corresponding text queries
+                    boxes, scores, labels = results[i]["boxes"], results[i]["scores"], results[i]["labels"]
+                    for box, score, label in zip(boxes, scores, labels):
+                        box = [round(i, 2) for i in box.tolist()]
+                        print(f"Detected {text[label]} with confidence {round(score.item(), 3)} at location {box}")
```

Comment:
There is still an issue here because the outputs I get are:
```
Detected a with confidence 0.68 at location [324.0, 21.38, 640.0, 372.0]
Detected a with confidence 0.749 at location [2.03, 57.25, 316.0, 472.0]
```
while it should be
```
Detected a photo of a cat with confidence 0.68 at location [324.0, 21.38, 640.0, 372.0]
Detected a photo of a dog with confidence 0.749 at location [2.03, 57.25, 316.0, 472.0]
```
no?

Reply:
Nice catch. Yes it should be printing both cats.

Detected a **photo of a cat** with confidence 0.68 at location [324.0, 21.38, 640.0, 372.0]
Detected a **photo of a cat** with confidence 0.749 at location [2.03, 57.25, 316.0, 472.0]

The new submitted codes have passed the 'make style'